doc/tutorials/intermediate/create_form.md

```diff
@@ -0,0 +1,261 @@
+# Create a Form with HoloViz Panel
+
+Welcome to the "Create a Form" tutorial! In this guide, we will create an interactive and fully functional *form* using HoloViz Panel. By the end of this tutorial, you'll have built a form with validation that captures user input and processes it effectively. This tutorial adopts the `param.Parameterized` approach to ensure the form is extensible and maintainable.
+
+<iframe src="https://panel-org-create-form.hf.space" frameborder="0" style="width: 100%;height:500px"></iframe>
+
+---
+
+## Key Objectives
+
+- **Create an Interactive Form**: Build a form to capture Name, Email, and an optional Message.
+- **Validate User Input**: Ensure the form inputs meet specified criteria.
+- **Submit Data**: Simulate sending the form data for further processing.
+
+---
+
+## Implementation
+
+Below is a step-by-step breakdown of the code to create the form.
+
+### 1. Import Dependencies
+
+```python
+import panel as pn
+import param
+```
+
+We use `panel` to build the interactive components and `param` for structured data handling and validation.
+
+### 2. Define Form Text and Icons
+
+We add some static elements to make the form visually appealing.
+
+```python
+FORM_TEXT = """\
+<h1>Join Newsletter</h1>
+
+Get the latest updates and news about Panel.
+"""
+
+FORM_ICON = """\
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="2">
+  <path d="M3 7a2 2 0 0 1 2 -2h14a2 2 0 0 1 2 2v10a2 2 0 0 1 -2 2h-14a2 2 0 0 1 -2 -2v-10z"></path>
+  <path d="M3 7l9 6l9 -6"></path>
+</svg>
+"""
+```
+
+### 3. Create a Parameterized Form State Class
+
+The `FormState` class handles form inputs, validation, and state management.
+
+```python
+class FormState(param.Parameterized):
+    name = param.String(default="", doc="The name of the user.")
+    email = param.String(default="", doc="The email of the user.")
```

Comment:
It would be nice if we could add a regex for an `email`. I tried but the exceptions where raised in the terminal. It was not clear to me how to easily integrate them with the components and UI.

It would also be nice to be able to add a custom `validate` function to any `param.Parameter`. This seems not supported either. Its a standard thing for all other Model frameworks I know.

Reply:
I think regex works as long as you don't use `name` as your param name because the name property overrides the param.

<img width="816" alt="image" src="https://github.com/user-attachments/assets/7924821a-832a-4d4a-bf0a-ac0be0a51587" />


```
import param
class FormState(param.Parameterized):
    n = param.String(default="", doc="The name of the user.", regex=r"^[A-Za-z ]*$")


form_state = FormState(n="ABC")
form_state.name
```

Reply:
Thanks. The problem is not the `regex` not working. The problem is that the `regex` errors are not propogated to the UI. It stays on the server side.